transport/tcp/serve_tcp_ipmap.go

```diff
@@ -80,10 +88,15 @@ func newIPMapEntry(input string, ident string) (*ipMapEntry, error) {
 		}, nil
 	}
 
+	// expect CIDR and '*' placeholder expansion
+
 	if strings.Count(ident, "*") != 1 {
 		err = fmt.Errorf("CIDRs require 1 IP placeholder")
-		return nil, errors.Wrapf(err, "invalid client identity for IP %q", ip)
+		return nil, errors.Wrapf(err, "invalid client identity %q for IP %q", ident, ip)
 	}
+
+	// FIXME use longest possible validation string
```

Comment:
@problame 

What stops us from getting the root_fs and validating the length of the entire dataset name against  zfs.MaxDatasetNameLen?   Do we ever validate MaxLength of the entire ZFS dataset (root_fs + "/" + zfsComponent) anywhere else?

Reply:
>  Do we ever validate MaxLength of the entire ZFS dataset (root_fs + "/" + zfsComponent) anywhere else?

The sending-side filesystem name is unknown at the time we check the receiving-side config, thus anything we do will only be a best-guess.

This FIXME aims at using a maximum-length IP address string.

Reply:
@problame  Ahhhh that makes sense.    I guess any errors for that will show up in the log and zrepl status command?

Reply:
yes, exactly, it's a runtime error